### PR TITLE
fix(repoinfo): fix userprefs values for repos returned by expanded re…

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
…po info

- now isBookmarked and isStarred are updated correctly

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/test/mocks/repo_db_mock.go [hunk 1]
```diff
@@ -155,6 +157,14 @@ func (sdm RepoDBMock) GetRepoMeta(repo string) (repodb.RepoMetadata, error) {
 	return repodb.RepoMetadata{}, nil
 }
 
+func (sdm RepoDBMock) GetUserRepoMeta(ctx context.Context, repo string) (repodb.RepoMetadata, error) {
+	if sdm.GetRepoMetaFn != nil {
```
Comment (andaaron) [inline pkg/test/mocks/repo_db_mock.go:161]:
```suggestion
	if sdm.GetUserRepoMetaFn!= nil {
```

File: pkg/extensions/search/userprefs_test.go [hunk 2]
```diff
@@ -806,6 +806,97 @@ func TestGlobalSearchWithUserPrefFiltering(t *testing.T) {
 	})
 }
 
+func TestExpandedRepoInfoWithUserPrefs(t *testing.T) {
+	Convey("ExpandedRepoInfo with User Prefs", t, func() {
```
Comment (andaaron) [inline pkg/extensions/search/userprefs_test.go:810]: Maybe there should be also a repo which is not starred/bookmarked, and we test that as well.

File: pkg/extensions/search/resolver.go [hunk 3]
```diff
@@ -1096,13 +1096,26 @@ func expandedRepoInfo(ctx context.Context, repo string, repoDB repodb.RepoDB, cv
...
+	if err != nil {
+		return &gql_generated.RepoInfo{}, err
+	}
+
+	userBookmarkedRepos, err := repoDB.GetBookmarkedRepos(ctx)
+	if err != nil {
+		return &gql_generated.RepoInfo{}, err
+	}
+
+	repoMeta.IsStarred = zcommon.Contains(userStarredRepos, repo)
```
Comment (andaaron) [inline pkg/extensions/search/resolver.go:1116]: Is this still needed? GetUserRepoMeta should already cover this?

File: pkg/extensions/search/userprefs_test.go (hunk 2 above)
Comment (laurentiuNiculae) [reply pkg/extensions/search/userprefs_test.go:810]: I have added exhaustive tests for all cases

**Review by andaaron — APPROVED** [review]

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**